app/models/shortener/shortened_url.rb

```diff
@@ -136,7 +136,11 @@ def increment_usage_count
 
   def generate_unique_key
     charset = ::Shortener.key_chars
-    (0...::Shortener.unique_key_length).map{ charset[rand(charset.size)] }.join
+    key = nil
+    forbidden_keys_as_regex = /#{::Shortener.forbidden_keys.join("|")}/i
+    until key && (key =~ forbidden_keys_as_regex).present?
```

Comment:
The way this reads at first glance is `keep generating until a forbidden key *is* present`, but maybe I'm reading it wrong

Reply:
Yeah it's probably not super clear; it's because `key =~ forbidden_keys_as_regex` returns the index of the first matched character, so `.present?` basically checks it's not `nil`. I was going to use `nil?` with a bang but the double negation was confusing the hell out of me.

Reply:
Turns out it's because it should be the other way 🤦‍♂️